### ISSUE TITLE
Some genes lack 'exon' annotations after GFF3 to GTF conversion

### ISSUE THREAD

**Issue by wflynny** [gpertea/gffread#55]
Hello, I'm working on converting [several GFF3 files](https://stickleback.genetics.uga.edu/downloadData/) to GTF using `gffread`.  Some transcripts have their GFF3 `CDS` annotations converted to `exon` in the GTF file, while others don't.

Here are two small examples:
```
> cat tsen34.gff3
chrI	ensembl	gene	51956	53430	.	-	.	ID=ENSGACG00000004800;Name=tsen34;protein_id=ENSGACP00000006337
chrI	ensembl	mRNA	51956	53430	.	-	.	ID=ENSGACT00000006354;Parent=ENSGACG00000004800;Name=tsen34-201
chrI	ensembl	three_prime_UTR	51956	52039	.	-	.	ID=ENSGACT00000006354.UTR3_1;Parent=ENSGACT00000006354
chrI	ensembl	CDS	52040	52224	.	-	2	ID=ENSGACT00000006354.CDS_5;Parent=ENSGACT00000006354
chrI	ensembl	CDS	52623	52898	.	-	2	ID=ENSGACT00000006354.CDS_4;Parent=ENSGACT00000006354
chrI	ensembl	CDS	52988	53130	.	-	1	ID=ENSGACT00000006354.CDS_3;Parent=ENSGACT00000006354
chrI	ensembl	CDS	53151	53247	.	-	2	ID=ENSGACT00000006354.CDS_2;Parent=ENSGACT00000006354
chrI	ensembl	CDS	53265	53430	.	-	0	ID=ENSGACT00000006354.CDS_1;Parent=ENSGACT00000006354
> singularity exec gffread-0.11.8.sif gffread -T -o tsen34.gtf tsen34.gff3
> cat tsen34.gtf
chrI	ensembl	transcript	51956	53430	.	-	.	transcript_id "ENSGACT00000006354"; gene_id "ENSGACG00000004800";
chrI	ensembl	exon	51956	52224	.	-	.	transcript_id "ENSGACT00000006354"; gene_id "ENSGACG00000004800";
chrI	ensembl	exon	52623	52898	.	-	.	transcript_id "ENSGACT00000006354"; gene_id "ENSGACG00000004800";
chrI	ensembl	exon	52988	53130	.	-	.	transcript_id "ENSGACT00000006354"; gene_id "ENSGACG00000004800";
chrI	ensembl	exon	53151	53247	.	-	.	transcript_id "ENSGACT00000006354"; gene_id "ENSGACG00000004800";
chrI	ensembl	exon	53265	53430	.	-	.	transcript_id "ENSGACT00000006354"; gene_id "ENSGACG00000004800";
chrI	ensembl	CDS	52040	52224	.	-	2	transcript_id "ENSGACT00000006354"; gene_id "ENSGACG00000004800";
chrI	ensembl	CDS	52623	52898	.	-	2	transcript_id "ENSGACT00000006354"; gene_id "ENSGACG00000004800";
chrI	ensembl	CDS	52988	53130	.	-	1	transcript_id "ENSGACT00000006354"; gene_id "ENSGACG00000004800";
chrI	ensembl	CDS	53151	53247	.	-	2	transcript_id "ENSGACT00000006354"; gene_id "ENSGACG00000004800";
chrI	ensembl	CDS	53265	53430	.	-	0	transcript_id "ENSGACT00000006354"; gene_id "ENSGACG00000004800";
```
That works.  However, the following doesn't
```
> cat cox2.gff3
chrM	insdc	gene	7173	7863	.	+	.	ID=ENSGACG00000020938;Name=mt-co2;protein_id=ENSGACP00000027663
chrM	insdc	mRNA	7173	7863	.	+	.	ID=ENSGACT00000027730;Parent=ENSGACG00000020938;Name=mt-co2-201
chrM	insdc	CDS	7173	7863	.	+	0	ID=ENSGACT00000027730.CDS_1;Parent=ENSGACT00000027730
> singularity exec gffread-0.11.8.sif gffread -T -o cox2.gtf cox2.gff3
> cat cox2.gtf
chrM	insdc	transcript	7173	7863	.	+	.	transcript_id "ENSGACT00000027730"; gene_id "ENSGACG00000020938"; Name "mt-co2-201";
chrM	insdc	CDS	7173	7863	.	+	0	transcript_id "ENSGACT00000027730"; gene_id "ENSGACG00000020938";
```
Any clue why?  There are roughly 27,000 genes in GFF3 file and maybe 3000 or so have this problem.

Thanks

**Comment by gpertea** [comment]
Sorry for the late reply. It looks like the exon features are only added when there are UTR regions defined in the file. Otherwise the CDS features are identical to the exon features, right? The default mode of operation of gffread is to print the exon and CDS features as needed (and close to what it got as input). 

However there is a `--force-exons` that you can use to make gffread output exons too, in such cases. Unfortunately that option was not really working for GTF output (`-T` option), but I fixed that in the latest version of gffread which I just released (v0.12.1). With that version you can use the options `-T --force-exons` to get the output you wanted.

**Comment by wflynny** [comment]
Awesome, that makes sense.  Thanks for your help @gpertea.
